web-core-sdk/one-key-flow/single-page-example/src/App.tsx

```diff
@@ -0,0 +1,373 @@
+    const { torusNodeEndpoints, torusNodePub, torusIndexes } = await nodeDetailManager.getNodeDetails({ verifier, verifierId: sub });
+    const userDetails = await torus.getUserTypeAndAddress(torusNodeEndpoints, torusNodePub,  { verifier, verifierId: sub }, true);
+
+    // check if the user hasn't enabled one key login
+    if (userDetails.typeOfUser === "v2" && !userDetails.upgraded) {
+      // if YES, login directly with the torus libraries within your app
+      const keyDetails =  await torus.retrieveShares(torusNodeEndpoints, torusIndexes, verifier, { verifier_id: sub }, idToken, {});
+      // use the private key to get the provider
+      const finalPrivKey = subkey(keyDetails.privKey.padStart(64, "0"), Buffer.from(clientId, "base64")).padStart(64, "0");
+      const ethereumPrivateKeyProvider = new EthereumPrivateKeyProvider({
+        config: {
+          chainConfig,
+        },
+      });
+      await ethereumPrivateKeyProvider.setupProvider(finalPrivKey);
+      setProvider(ethereumPrivateKeyProvider.provider);
+      setUsesTorus(true);
+    } else {
+      // if NO, login with web3auth
+      const web3authProvider = await web3auth.connectTo(
+        WALLET_ADAPTERS.OPENLOGIN,
+        {
+          loginProvider: "jwt",
+          extraLoginOptions: {
+            id_token: idToken,
+            verifierIdField: "sub",
+            domain: "http://localhost:3000",
+          },
+        }
+      );
+      setProvider(web3authProvider);
+      setUsesTorus(false);
+    }
+  };
+
+  const getUserInfo = async () => {
+    if (!web3auth) {
+      uiConsole("web3auth not initialized yet");
+      return;
+    }
+    if (usesTorus) {
+      uiConsole("You are directly using Torus Libraries to login the user, hence the Web3Auth <code>getUserInfo</code> function won't work for you. Get the user details directly from id token.", parseToken(idToken));
+      return;
+    }
+    const user = await web3auth.getUserInfo();
+    uiConsole(user);
+  };
+
+  const authenticateUser = async () => {
+    if (!web3auth) {
+      uiConsole("web3auth not initialized yet");
+      return;
+    }
+    if (usesTorus) {
+      uiConsole("You are directly using Torus Libraries to login the user, hence the Web3Auth <code>authenticateUser</code> function won't work for you. For server side verification, directly use your login provider and id token. <br/><br/> You can use the Firebase Id Token for example in this case. <br/>", idToken);
+      return;
+    }
+    const id_token = await web3auth.authenticateUser();
+    // console.log(JSON.stringify(user, null, 2))
+    uiConsole("Id Token:", id_token, "You can use this id token from Web3Auth for server side verification from your own end, visit the web3auth documentation for more information.");
+  };
+
+  const logout = async () => {
+    if (!web3auth) {
+      uiConsole("web3auth not initialized yet");
+      return;
+    }
+    if (usesTorus) {
```

Comment:
Can we show logout with Firebase here? because this guide does use Firebase. It would be good to have logout from Firebase here.